src/Core/Style.js

```diff
@@ -81,7 +90,34 @@ function getImage(source, key) {
     const target = document.createElement('img');
 
     if (typeof source == 'string') {
-        target.src = source;
+        const color = key;
+        if (color) {
+            Fetcher.texture(source, { crossOrigin: 'anonymous' })
+                .then(texture => texture.image)
+                .then((img) => {
+                    canvas.width = img.naturalWidth;
+                    canvas.height = img.naturalHeight;
+                    const ctx = canvas.getContext('2d', { willReadFrequently: true });
+                    ctx.drawImage(img, 0, 0);
+                    const imgd = ctx.getImageData(0, 0, img.naturalWidth, img.naturalHeight);
+
+                    const pix = imgd.data;
+
+                    const colorRGB = hex2rgb(color);
+                    for (var i = 0, n = pix.length; i < n; i += 4) {
+                        if (pix[i] + pix[i + 1] + pix[i + 2] === 255 * 3) {
+                            pix[i] = colorRGB.r;
+                            pix[i + 1] = colorRGB.g;
+                            pix[i + 2] = colorRGB.b;
+                            // pix[i+3] is the transparency.
+                        }
+                    }
```

Comment:
there's a little bug with icon border

Reply:
It's not really a bug as the remaining pixel aren't white (but seems like white around colored pixel). We can say that the icon image is not clean enought.
If needed as some point it will be able to be changed.

Reply:
I propose a icon color refactoring in this commit https://github.com/iTowns/itowns/commit/5510ae13df711c9ae0eac6b3bc2aec49225ac3a9
It uses delta-e to replace color.